packages/panna-sdk/src/core/utils/activity.ts

```diff
@@ -349,21 +449,38 @@ export const getActivitiesByAddress = async function (
               (e) => e.token.type.toLowerCase() === TokenERC.ERC20
             );
             const erc20TxTotal = erc20Tx?.total as BlockscoutTotalERC20;
+            const decimals = erc20Tx?.token.decimals
+              ? Number(erc20Tx?.token.decimals)
+              : 0;
 
-            amount = {
+            const erc20Amount: ERC20Amount = {
               type: TokenERC.ERC20,
               value: erc20TxTotal.value,
               tokenInfo: {
-                address: erc20Tx?.token.address_hash || erc20Tx?.token.address,
-                name: erc20Tx?.token.name,
-                symbol: erc20Tx?.token.symbol,
-                decimals: erc20Tx?.token.decimals
-                  ? Number(erc20Tx?.token.decimals)
-                  : 0,
+                address:
+                  (erc20Tx?.token.address_hash || erc20Tx?.token.address) ?? '',
+                name: erc20Tx?.token.name ?? '',
+                symbol: erc20Tx?.token.symbol ?? '',
+                decimals,
                 type: TokenERC.ERC20,
-                icon: erc20Tx?.token.icon_url
+                icon: erc20Tx?.token.icon_url ?? null
```

Comment:
The defaults can be avoided if we extract erc20TxToken as below:
```
const erc20TxToken = erc20Tx?.token as BlockscoutTokenInfo;
...
  name: erc20TxToken.name
```

Reply:
Done.